src/main/java/hudson/plugins/ws_cleanup/Wipeout.java

```diff
@@ -61,23 +69,46 @@ protected void perform(FilePath workspace) throws IOException, InterruptedExcept
                     new Object[] { workspace.getRemote(), deleteMe.getName() }
             );
             workspace.deleteRecursive();
+        } else {
+            String slaveName = toComputer(deleteMe.getChannel()).getName();
```

Comment:
The only situation I can imagine when `FilePath` representing workspace in a build step is backed up by a channel that does not belong to any computer is that the computer got disconnected in the middle of the build. Or perhaps someone is reusing these classes for something nasty. Anyway, getting back to synchronous deletion will either work or fail with an exception resembling the cause. Will fix.